### PR TITLE
Move sort order preferences to AppPreferences interface

### PR DESCRIPTION
**Pull request by ezaquarii**
Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

### CODE REVIEW

File: src/main/java/com/owncloud/android/ui/preview/PreviewImagePagerAdapter.java [hunk 1]
```diff
@@ -56,6 +57,7 @@
     private Set<Integer> mDownloadErrors;
     private FileDataStorageManager mStorageManager;
     private SparseArray<FileFragment> mCachedFragments;
+    private AppPreferences preferences;
```
[the remark this answers is not in the document]
Comment (ezaquarii) [reply src/main/java/com/owncloud/android/ui/preview/PreviewImagePagerAdapter.java]: Fixed.

File: src/main/java/com/owncloud/android/ui/preview/PreviewImagePagerAdapter.java [hunk 2]
```diff
@@ -67,7 +69,7 @@
      */
     public PreviewImagePagerAdapter(FragmentManager fragmentManager, OCFile parentFolder,
                                     Account account, FileDataStorageManager storageManager,
-                                    boolean onlyOnDevice, Context context) {
+                                    boolean onlyOnDevice, AppPreferences preferences, Context context) {
```
[the remark this answers is not in the document]
Comment (ezaquarii) [reply src/main/java/com/owncloud/android/ui/preview/PreviewImagePagerAdapter.java]: Fixed.

File: src/main/java/com/nextcloud/client/preferences/AppPreferences.java [hunk 3]
```diff
@@ -20,10 +20,45 @@
 
 package com.nextcloud.client.preferences;
 
+import android.content.Context;
```
[the remark this answers is not in the document]
Comment (ezaquarii) [reply src/main/java/com/nextcloud/client/preferences/AppPreferences.java]: Fixed.

File: src/main/java/com/owncloud/android/ui/preview/PreviewImagePagerAdapter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ezaquarii) [reply src/main/java/com/owncloud/android/ui/preview/PreviewImagePagerAdapter.java]: Fixed.

**Comment by ezaquarii** [comment]
This needs some more refactoring around `ComponentsGetter` interface as it crashes at runtime.

**Comment by ezaquarii** [comment]
This needs some fixes around `ComponentsGetter` as it crashes the app at runtime.

**Comment by ezaquarii** [comment]
Integration test failed due to cancellation. Otherwise compile & test are green.

https://drone.nextcloud.com/nextcloud/android/8182/6

Not sure what when wrong here.

File: src/main/java/com/nextcloud/client/preferences/PreferenceManager.java [hunk 4]
```diff
@@ -342,11 +324,7 @@ public static String getFolderPreference(Context context, String preferenceName,
         }
 
         ArbitraryDataProvider dataProvider = new ArbitraryDataProvider(context.getContentResolver());
-        FileDataStorageManager storageManager = ((ComponentsGetter)context).getStorageManager();
```
[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply src/main/java/com/nextcloud/client/preferences/PreferenceManager.java]: I guess the reason to re-use storageManager is that it cheaper than re-creating it. There should be no unexpected behaviours as we always directly store values/content.

But creating it directly would also eliminiate the next null check.

Totally agree with injecting!

**Comment by tobiasKaminsky** [comment]
> Integration test failed due to cancellation.

Sometimes if the host is out of memory, build will be cancelled, but this happens very rarely.

**Comment by ezaquarii** [comment]
It's cancelled again:

```
+ git init
Initialized empty Git repository in /drone/src/github.com/nextcloud/android/.git/
+ git remote add origin https://github.com/nextcloud/android.git
+ git fetch --no-tags origin +refs/pull/3713/merge:
From https://github.com/nextcloud/android
 * branch                refs/pull/3713/merge -> FETCH_HEAD
+ git checkout -qf FETCH_HEAD
+ git submodule update --init --recursive
```

It is a bit funny that this is the only PR that gives me those strange issues. I'll rebase and try again.

**Comment by ezaquarii** [comment]
@tobiasKaminsky @AndyScherzinger It looks like all checks are passed now and I fixed all pending issues. Can we merge it now?

**Comment by AndyScherzinger** [comment]
Sure thing @ezaquarii 👍 I've been only on the phone only during the weekend and have to travel to my client's office on Mondays (until around noon... 🎉)

**Pull request merged**